packages/browser/src/tracing/linkedTraces.ts

```diff
@@ -0,0 +1,239 @@
+import type { Client, PropagationContext, Span } from '@sentry/core';
+import {
+  type SpanContextData,
+  getCurrentScope,
+  getRootSpan,
+  logger,
+  SEMANTIC_ATTRIBUTE_SENTRY_PREVIOUS_TRACE_SAMPLE_RATE,
+  SEMANTIC_ATTRIBUTE_SENTRY_SAMPLE_RATE,
+  SEMANTIC_LINK_ATTRIBUTE_LINK_TYPE,
+  spanToJSON,
+} from '@sentry/core';
+import { DEBUG_BUILD } from '../debug-build';
+import { WINDOW } from '../exports';
+
+export interface PreviousTraceInfo {
+  /**
+   * Span context of the previous trace's local root span
+   */
+  spanContext: SpanContextData;
+
+  /**
+   * Timestamp in seconds when the previous trace was started
+   */
+  startTimestamp: number;
+
+  /**
+   * sample rate of the previous trace
+   */
+  sampleRate: number;
+
+  /**
+   * The sample rand of the previous trace
+   */
+  sampleRand: number;
+}
+
+// 1h in seconds
+export const PREVIOUS_TRACE_MAX_DURATION = 3600;
+
+// session storage key
+export const PREVIOUS_TRACE_KEY = 'sentry_previous_trace';
+
+export const PREVIOUS_TRACE_TMP_SPAN_ATTRIBUTE = 'sentry.previous_trace';
+
+/**
+ * Takes care of linking traces and applying the (consistent) sampling behavoiour based on the passed options
+ * @param options - options for linking traces and consistent trace sampling (@see BrowserTracingOptions)
+ * @param client - Sentry client
+ */
+export function linkTraces(
```

Comment:
A minor thing (feel free to ignore): You could switch the arguments here to have client first and the options second. As the options might get quite verbose the client is easy to miss when looking at the code.

Reply:
Good call, agreed, options should be after the client :)